aiida/cmdline/commands/cmd_rabbitmq.py

```diff
@@ -0,0 +1,386 @@
+#                                                                         #
+# The code is hosted on GitHub at https://github.com/aiidateam/aiida-core #
+# For further information on the license, see the LICENSE.txt file        #
+# For further information please visit http://www.aiida.net               #
+###########################################################################
+"""`verdi devel rabbitmq` commands."""
+from __future__ import annotations
+
+import collections
+import re
+import sys
+import typing as t
+
+import click
+import requests
+import tabulate
+import wrapt
+import yaml
+
+from aiida.cmdline.commands.cmd_devel import verdi_devel
+from aiida.cmdline.params import arguments, options
+from aiida.cmdline.utils import decorators, echo
+
+if t.TYPE_CHECKING:
+    import kiwipy.rmq
+
+    from aiida.manage.configuration.profile import Profile
+
+
+@verdi_devel.group('rabbitmq')
+def cmd_rabbitmq():
+    """Commands to interact with RabbitMQ."""
+
+
+@cmd_rabbitmq.group('queues')
+def cmd_queues():
+    """Commands to interact with RabbitMQ queues."""
+
+
+@cmd_rabbitmq.group('tasks')
+def cmd_tasks():
+    """Commands related to process tasks."""
+
+
+available_projectors = (
+    'arguments',
+    'auto_delete',
+    'backing_queue_status',
+    'consumer_utilisation',
+    'consumers',
+    'durable',
+    'effective_policy_definition',
+    'exclusive',
+    'exclusive_consumer_tag',
+    'garbage_collection',
+    'head_message_timestamp',
+    'idle_since',
+    'memory',
+    'message_bytes',
+    'message_bytes_paged_out',
+    'message_bytes_persistent',
+    'message_bytes_ram',
+    'message_bytes_ready',
+    'message_bytes_unacknowledged',
+    'messages',
+    'messages_details',
+    'messages_paged_out',
+    'messages_persistent',
+    'messages_ram',
+    'messages_ready',
+    'messages_ready_details',
+    'messages_ready_ram',
+    'messages_unacknowledged',
+    'messages_unacknowledged_details',
+    'messages_unacknowledged_ram',
+    'name',
+    'node',
+    'operator_policy',
+    'policy',
+    'recoverable_slaves',
+    'reductions',
+    'reductions_details',
+    'single_active_consumer_tag',
+    'state',
+    'type',
+    'vhost',
+)
+
+
+def echo_response(response: requests.Response, exit_on_error: bool = True) -> None:
+    """Echo the response of a request.
+
+    :param response: The response to the request.
+    :param exit_on_error: Boolean, if ``True``, call ``sys.exit`` with the status code of the response.
+    """
+    try:
+        response.raise_for_status()
+    except requests.HTTPError:
+        click.secho(f'[{response.status_code}] ', fg='red', bold=True, nl=False)
+        click.secho(f'{response.reason}: {response.url}')
+        if exit_on_error:
+            sys.exit(response.status_code)
+    else:
+        click.secho(f'[{response.status_code}] ', fg='green', bold=True)
+
+
+@wrapt.decorator
+@click.pass_context
+def with_client(ctx, wrapped, _, args, kwargs):
```

Comment:
a significant portion of the code in this file seems quite independent of aiida and could possibly fit inside, say, kiwipy as well.

I'm not saying this has to happen, just worth consideration.
We may also want to comment in the module docstring in why these tools are needed, and under which circumstances we may want to remove them again (if that is the case).

Reply:
Fair, except for the `tasks analysis` command though, that is purely AiiDA specific. But you are right, the rest could be abstracted (currently depends tightly on AiiDA's `Manager` to get the communicator) and be put in `kiwipy`. For me would be fine to move it there at some point if we find it is useful more generically. I had a quick search and there are a few Python libraries out there that provide an interface to the HTTP API, but they are not well maintained.